x-pack/plugins/infra/public/components/infrastructure_node_metrics_tables/shared/hooks/metrics_to_api_options.ts

```diff
@@ -57,27 +68,43 @@ export interface NodeMetricsExplorerOptionsMetric<Field extends string>
   field: Field;
 }
 
-export function metricsToApiOptions<T extends string>(metricsMap: MetricsMap<T>, groupBy: string) {
-  const metrics = Object.values(metricsMap) as Array<NodeMetricsExplorerOptionsMetric<T>>;
+export function metricsToApiOptions<T extends string>(
+  metricsQueryOptions: MetricsQueryOptions<T>,
+  filterClauseDsl?: QueryDslQueryContainer
+) {
+  const metrics = Object.values(metricsQueryOptions.metricsMap) as Array<
+    NodeMetricsExplorerOptionsMetric<T>
+  >;
 
   const options: MetricsExplorerOptions = {
     aggregation: 'avg',
-    groupBy,
+    groupBy: metricsQueryOptions.groupByField,
     metrics,
+    filterQuery: JSON.stringify(
+      buildFilterClause(metricsQueryOptions.sourceFilter, filterClauseDsl)
+    ),
   };
 
-  const metricByField = createFieldLookup(Object.keys(metricsMap) as T[], metrics);
-
   return {
     options,
-    metricByField,
   };
 }
 
-function createFieldLookup<T extends string>(
-  fields: T[],
-  metrics: Array<NodeMetricsExplorerOptionsMetric<T>>
-) {
+function buildFilterClause(
+  sourceFilter: QueryDslQueryContainer,
+  filterClauseDsl?: QueryDslQueryContainer
+): QueryDslQueryContainer {
+  return {
+    bool: {
+      filter: !!filterClauseDsl ? [sourceFilter, filterClauseDsl] : [sourceFilter],
+    },
+  };
```

Comment:
```suggestion
  const filter = [sourceFilter];
  
  if (filterClauseDsl) {
    filter.push(filterClauseDsl);
  }
  
  return {
    bool: {
      filter,
    },
  };
```

NIt: I'm just allergic to ternaries for the sake of readability.

Reply:
I tend to agree with that. But here it doesn't seem to be harming the readability. But it's my personal opinion. I could apply your suggestion it if it hurting you eyes :)!

Reply:
The mutation hurts the soul, not the eyes :D 
I'm fine either way @crespocarlos, as long as I get to complain about ternaries :)